File: src/main/java/com/ascargon/rocketshow/update/UpdateState.java
package com.ascargon.rocketshow.update;

import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;


@XmlRootElement
@Getter
@Setter
public class UpdateState {

    private UpdateStep step;
    private String updatingFromVersion;
    private double progressPercentage;
    private String progressMessage;
    private String error;

}

File: src/main/webapp/src/app/models/update-state.ts
export class UpdateState {
    step: string;
    updatingFromVersion: string;
    progressPercentage: number;
    progressMessage: string;
    error: string;

    constructor(data?: any) {
        if (!data) {
            return;
        }

        this.step = data.step;
        this.updatingFromVersion = data.updatingFromVersion;
        this.progressPercentage = data.progressPercentage;
        this.progressMessage = data.progressMessage;
        this.error = data.error;
    }
}
